CryptKeeperZK/crypt-keeper-extension
# feat: add validation for onboarding page

## Explanation

This PR adds password strength rules for onboarding page and some testing chores.

Details are below:
- [x] Add password strength rules
- [x] Integrate yup and react-hook-form
- [x] Add missing tests for Home and Header
- [x] Set coverage threshold

## More Information

Closes #191
Blocked by #200 

## Screenshots/Screencaps

### Before

N/A

### After

![image](https://user-images.githubusercontent.com/14254374/228951903-1097bf62-e38e-46b3-a6cc-086392b81133.png)
![image](https://user-images.githubusercontent.com/14254374/228951963-edb685bb-b124-4e7d-90b1-9b2826823040.png)

## Manual Testing Steps

1. Go to onboarding page
2. Try to check different types and combinations of passwords and then submit the form
3. Check validation messages

## Pre-Merge Checklist


### + If there are functional changes:


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
